pkg/framework/plugins/nodeutilization/normalizer/normalizer.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package normalizer
+
+import (
+	"math"
+
+	"golang.org/x/exp/constraints"
+)
+
+// Normalizer is a function that receives two values of the same type and
+// return an object of a different type. An usage case can be a function
+// that converts a memory usage from mb to % (the first argument would be
+// the memory usage in mb and the second argument would be the total memory
+// available in mb).
+type Normalizer[V, N any] func(V, V) N
+
+// Indexer is a function that receives an object and returns a key. This
+// function is used to generate a key for a map based on the object.
+type Indexer[K comparable, O any] func(O) K
+
+// Values is a map of values indexed by a comparable key. An example of this
+// can be a list of resources indexed by a node name.
+type Values[K comparable, V any] map[K]V
+
+// Number is an interface that represents a number. Represents things we
+// can do math operations on.
+type Number interface {
+	constraints.Integer | constraints.Float
+}
+
+// Normalize uses a Normalizer function to normalize a set of values. For
+// example one may want to convert a set of memory usages from mb to %.
+// This function receives a set of usages, a set of totals, and a Normalizer
+// function. The function will return a map with the normalized values.
+func Normalize[K comparable, V, N any](usages, totals Values[K, V], fn Normalizer[V, N]) map[K]N {
+	result := Values[K, N]{}
+	for key, value := range usages {
+		total, ok := totals[key]
+		if !ok {
+			continue
+		}
+		result[key] = fn(value, total)
+	}
+	return result
+}
+
+// Replicate replicates the provide value for each key in the provided slice.
+// Returns a map with the keys and the provided value.
+func Replicate[K comparable, V any](keys []K, value V) map[K]V {
+	result := map[K]V{}
+	for _, key := range keys {
+		result[key] = value
+	}
+	return result
+}
+
+// Min receives a minimum value and a list of values. Returns a list of values
+// where each value is greater than the minimum value.
+func Min[K comparable, N Number, V ~map[K]N](values []V, limit N) []V {
```

Comment:
Better to keep the function as simple as possible. I.e. `(values V, limit N) V` and:
```
items := 		normalizer.Deviate(
				normalizer.Average(usage),
				[]api.ResourceThresholds{
					lowSpan.Negative(),
					highSpan,
				},
			),
[]api.ResourceThresholds{
	normalizer.Min(items[0], 1),
	normalizer.Min(items[1], 1),
},
```
I know I mentioned it will be great to be able to chain the operators. In this case introducing a `map` function would solve it:
```
	map([]api.ResourceThresholds{
			normalizer.Sum(normalizer.Average(usage), lowSpan.Negative()),
			normalizer.Sum(normalizer.Average(usage), highSpan),
		}, 
		func(item api.ResourceThresholds) { return normalizer.Min(item, 1) }
	)
```